front/src/applications/operationalStudies/hooks/usePathProjection.ts

```diff
@@ -4,19 +4,26 @@ import { useSelector } from 'react-redux';
 
 import { osrdEditoastApi, type InfraWithState } from 'common/api/osrdEditoastApi';
 import usePathProperties from 'modules/pathfinding/hooks/usePathProperties';
+import type { TrainScheduleId } from 'reducers/osrdconf/types';
 import { getTrainIdUsedForProjection } from 'reducers/simulationResults/selectors';
+import { formatTrainScheduleIdToEditoastTrainId } from 'utils/trainId';
 
 const usePathProjection = (infra: InfraWithState) => {
   const trainIdUsedForProjection = useSelector(getTrainIdUsedForProjection);
 
+  // TODO Paced train : Adapt this to handle paced trains in issue https://github.com/OpenRailAssociation/osrd/issues/10615
+  const editoastTrainIdUsedForProjection = formatTrainScheduleIdToEditoastTrainId(
+    trainIdUsedForProjection as TrainScheduleId
+  );
+
   const { data: projectionPathResult } =
     osrdEditoastApi.endpoints.getTrainScheduleByIdPath.useQuery(
       {
-        id: trainIdUsedForProjection!,
+        id: editoastTrainIdUsedForProjection!,
         infraId: infra.id,
       },
       {
-        skip: !trainIdUsedForProjection,
+        skip: !editoastTrainIdUsedForProjection,
       }
     );
```

Comment:
I suppose that you made formatTrainScheduleIdToEditoastTrainId accept undefined value for this use case. It might be better to make it do not accept undefined value and rewrite the code like this, what do you think ? (I'm not 100% sure it will work with typescript)
```suggestion
  const { data: projectionPathResult } =
    osrdEditoastApi.endpoints.getTrainScheduleByIdPath.useQuery(
      {
        id: formatTrainScheduleIdToEditoastTrainId(trainIdUsedForProjection as TrainScheduleId),
        infraId: infra.id,
      },
      {
        skip: !trainIdUsedForProjection,
      }
    );
```

Reply:
Seems to be working, I will give it a try :)

Reply:
Can we avoid `as`, and use `!` instead?

Reply:
Done